File: 백준/Bronze/1934. 최소공배수/최소공배수.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Arrays;
import java.util.stream.Stream;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int T = Integer.parseInt(br.readLine());

        for (int i = 0; i < T; i++) {

            int[] input = Stream.of(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
            int big = Arrays.stream(input).max().getAsInt();
            int small = Arrays.stream(input).min().getAsInt();
            int num = 1;
            int result = 1;

            while (num <= small) {
                if (big * num % small == 0) {
                    result = big * num;
                    break;
                } else {
                    num++;
                }
            }
            bw.write(String.valueOf(result) + "\n");
        }
        bw.flush();
        bw.close();
    }
}

File: 백준/Bronze/1934. 최소공배수/최소공배수.py
from math import prod
import sys
input = sys.stdin.readline

T = int(input().rstrip())
for i in range(T):
    num = list(map(int, input().split()))
    big, small = max(num), min(num)
    divNum = small
    arr = []

    while divNum >= 2:
        if big % divNum == 0 and small % divNum == 0:
            arr.append(divNum)
            big //= divNum
            small //= divNum
            divNum = small
        else:
            divNum -= 1

    print(big * small * prod(arr))